### PR TITLE
GRAPHICS: correctly load all mipmaps for a TXB texture

### PR DESCRIPTION
**Pull request by farmboy0**

### CODE REVIEW

File: src/images/txb.cpp [hunk 1]
```diff
@@ -177,18 +159,27 @@ void TXB::readHeader(Common::SeekableReadStream &txb, byte &encoding) {
...
-			break;
+		if (width  > 1) width  >>= 1;
+		if (height > 1) height >>= 1;
 	}
 
 	if ((mipMapCount != 0) && (_mipMaps.empty()))
 		throw Common::Exception("Couldn't read any mip maps");
 }
 
+int TXB::computeMipMapSize(int format, int width, int height) {
```
Comment (DrMcCoy) [inline src/images/txb.cpp]: That does the same as getTXBDataSize(), no?

File: src/images/txb.cpp [hunk 2]
```diff
@@ -177,18 +159,27 @@ void TXB::readHeader(Common::SeekableReadStream &txb, byte &encoding) {
...
 		_mipMaps.push_back(mipMap.release());
 
-		width      >>= 1;
-		height     >>= 1;
-		mipMapSize >>= 2;
-
-		if ((width < 1) && (height < 1))
-			break;
+		if (width  > 1) width  >>= 1;
+		if (height > 1) height >>= 1;
```
Comment (DrMcCoy) [inline src/images/txb.cpp]: This now favours adding a whole lot of 1x1 mipmaps if the file specifies them. Does it really matter?

**Comment by farmboy0** [comment]
There is only one 1x1 mipmap in each texture.
There might be a 4x1 and 2x1 mipmap though depending on texture size.
And checking height and width before the shift allows for non rectangular textures to load all those mipmaps.
And no its not *really* important, just not accurate.
Indeed getTXBDataSize does the same.

File: src/images/txb.cpp (hunk 2 above)
Comment (farmboy0) [inline src/images/txb.cpp]: It now loads every mipmap the texture specifies and there is only every one 1x1 texture.
This happened before for rectangular mipmaps, now it also happens for non rectangular ones too.
Besides the code is much simpler and thus hopefully easier to read.

**Comment by DrMcCoy** [comment]
Thanks! I also applied the same patch to xoreos-tools and xoreos proper.

**Pull request merged**